dask/array/core.py

```diff
@@ -1337,6 +1337,10 @@ def __setitem__(self, key, value):
             self.dtype = y.dtype
             self.dask = y.dask
             self.name = y.name
+            # In order to figure out what keys to use, __dask_keys__ needs to
+            # know what chunks to expect. The chunks to expect are y.chunks, so
+            # we need to change our chunks.
```

Comment:
I recommend that we remove this comment

Reply:
This was not obvious to me, and this is... only place in the code where `_chunks` is set outside of `__new__`. Happy to remove it if you don't like it, though.

Reply:
Yeah, I'd prefer to remove it for now if that's ok